example/lib/develop/repository/peripheral_list_repository.dart

```diff
@@ -0,0 +1,49 @@
+import 'dart:async';
+
+import 'package:blemulator_example/develop/model/ble_peripheral.dart';
+import 'package:blemulator_example/example_peripheral.dart';
+import 'package:flutter_ble_lib/flutter_ble_lib.dart';
+import 'package:blemulator/blemulator.dart';
+
+typedef void ScanEventOutputFunction(BlePeripheral peripheral);
+
+class PeripheralListRepository {
```

Comment:
could you try to describe the responsibilities of the class?

Reply:
What about `PeripheralsRepository`?

Reply:
Repository's responsibility is to the serve as a data source for the bloc. So the idea behind it was to hide `BleManager` behind it, so bloc does not need to know what is responsible for serving the data. 

Apart from that repository would ultimately also be a place where we can decide if we want to switch from `Blemulator` to a real Bluetooth adapter.

Reply:
Renamed to `PeripheralsRepository`